lms/djangoapps/api_manager/courses/tests.py

```diff
@@ -2148,6 +2148,19 @@ def test_courses_data_metrics(self):
         response = self.do_get(course_metrics_uri)
         self.assertEqual(response.status_code, 404)
 
+    def test_course_data_metrics_user_group_filter(self):
```

Comment:
@ziafazal created separate test for each scenario and yes we are covering one user in multiple groups scenario to test distinct users.